fairlearn/postprocessing/_tradeoff_curve_utilities.py

```diff
@@ -139,17 +146,16 @@ def _filter_points_to_get_convex_hull(points_sorted):
             else:
                 break
         selected.append(r2)
-    return selected
+    return pd.DataFrame(selected)[[X_KEY, Y_KEY, OPERATION_KEY]]
 
 
 def _interpolate_curve(
     data: pd.DataFrame, x_col: str, y_col: str, content_col: str, x_grid: NDArray
-):
+) -> pd.DataFrame:
     """Interpolates the DataFrame in `data` along the values in `x_grid`.
 
-    Assumes: (1) data[y_col] is convex and non-decreasing in data[x_col]
-             (2) min and max in x_grid are below/above min and max in data[x_col]
-             (3) data is indexed 0,...,len(data)
+    Assumes: (1) data[y_col] is concave in data[x_col]
```

Comment:
The function should be concave, it is the shape made by the curve with the x-axis that is convex, if I'm not mistaken